python/ct/client/db/cert_desc.py

```diff
@@ -48,6 +50,27 @@ def from_cert(certificate, observations=[]):
         pass
 
     try:
+        tbs_alg = certificate.signature()["algorithm"]
+        proto.tbs_signature.algorithm_id = tbs_alg.long_name
+
+        tbs_params = certificate.signature()["parameters"]
+        proto.tbs_signature.parameters = str(tbs_params)
+
+        cert_alg = certificate.signature_algorithm()["algorithm"]
+        proto.cert_signature.algorithm_id = cert_alg.long_name
+
+        cert_params = certificate.signature_algorithm()["parameters"]
+        proto.cert_signature.parameters = str(cert_params)
+    except cert.CertificateError:
+        pass
+
+    try:
+        is_ca = certificate.basic_constraint_ca()
+        proto.basic_constraint_ca = is_ca is not None and is_ca.__bool__()
```

Comment:
the 'is not None' bit is superfluous. Also, if type(is_ca) is not bool, you'll get an AttributeError here.
Wouldn't you get the same result from this expression by using bool(is_ca) ?